src/services/utility-helpers.js

```diff
@@ -768,3 +768,112 @@ export const groupBy = (arry, key) => {
 export const sortAnnotations = (annotations) => {
   return annotations.sort((a, b) => a.time?.start - b.time?.start);
 };
+
+/**
+ * Truncates text that may contain HTML to a given maximum character length
+ * while preserving the HTML structure
+ * 
+ * @param {String} htmlString string to truncate which might contain HTML markup
+ * @param {Number} maxLength allowed max character length
+ * @returns {String} truncated HTML
+ */
+export const truncateText = (htmlString, maxLength) => {
+  if (htmlString.length <= maxLength) {
+    return { truncated: htmlString, hasShowMore: false };
+  }
+
+  // Create a temporary div to work with the HTML
+  const tempDiv = document.createElement('div');
+  tempDiv.innerHTML = htmlString;
+
+  const textLength = getTextLength(tempDiv);
+
+  if (textLength <= maxLength) {
```

Comment:
This is an edge case but what do you think about this change to account for the space taken by the ellipsis?  If a text is only 3 characters longer than the max then a user wouldn't have to click the show more button to see the the rest when it could have been rendered initially.

```suggestion
  if (textLength <= maxLength + 3) {
```